learn-array/src/transform.js

```diff
@@ -0,0 +1,15 @@
+function transform(data) {
+    var resultData = new Array;
+    var resultData = data.map(function (data) {
+        data.name = data.name.toUpperCase();
+        data.totalPoints = data.cards.reduce(function (accumulator, currentValue) {
+            debugger;
+            return accumulator + currentValue.point;
+        }, 0);
+        data.cards = data.cards;
+        return data;
```

Comment:
Better to give a different name inside the callback, duplication of the same name is not good.
```is
 data.map(function (item) {

})
```

Reply:
And instead of mutate the original item, should create a new object,
```js

 var resultData = data.map(function (data) {
	        return 
                {  name = data.name.toUpperCase(),
                  cards: data.cards,
	       totalPoints: data.cards.reduce(function (accumulator, currentValue) {....},
             }
    })
 ```    
	  

```